app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -190,6 +192,8 @@ public function getCartDataJs($checkoutType = 'multi-page')
                     $boltHelper->applyShippingRate($sessionQuote, $shippingMethod);
                 }
 
+            } else {
+                return '';
```

Comment:
@YevhenShBolt This is based from your change and without this, it is throwing an error.  I am very unfamiliar with the "Add Bolt Button Everywhere", so can you give a better explanation of what the problem is when using that new feature?  Otherwise, can you make the change that you are suggesting as a Pull Request to this branch?

From what I understand, you want to return a javascript alert.  My concern is that having a javascript alert pop up on the shopping cart page the already says *Shopping Cart is Empty* will be perceived as an intrusive Bolt introduced problem to most merchants.

My proposal is to make the *Add Bolt Button Everywhere* conform to receiving an empty string, as the point of `getCartDataJs` is to return an executable javascript string. and, if the quote is empty, we shouldn't be generating any cart JS.  If the quote is empty, then either it does not call 
`getCartDataJs`, or if it calls this method, and it receives an empty string, then the Bolt Button should not be rendered.

I assume the *Add Bolt Button Everywhere* button is the mini-cart and not an "Instant Buy" type button.

Reply:
@mccottry "add bolt button everywhere" is a mini-cart. For minicart, we want to keep showing bolt button and show alert when user clicks Bolt button.
If you look at existing code below, we have something like below, here we leverage `check` function to check isEmptyQuote flag and show alert.

```
                BoltCheckout.configure(
                    json_cart,
                    $jsonHints,
                    {
                      check: function() {           
                        $check
                        $checkForAdmin
                        if (isEmptyQuote) {
                            alert('{$boltHelper->__('Your shopping cart is empty. Please add products to the cart.')}');
                            return false;
                        }
                        if (!json_cart.orderToken) {
                            alert(json_cart.error);
                            return false;
                        }
                        return true;
                      },
```